src/uct/api/v2/uct_v2.h

```diff
@@ -1083,6 +1083,209 @@ ucs_status_t
+    UCT_IFACE_ATTR_FIELD_CAP_TAG_EAGER_MAX_IOV         = UCS_BIT(25),
+
+    UCT_IFACE_ATTR_FIELD_CAP_TAG_RNDV_MAX_ZCOPY        = UCS_BIT(26),
+    UCT_IFACE_ATTR_FIELD_CAP_TAG_RNDV_MAX_HDR          = UCS_BIT(27),
+    UCT_IFACE_ATTR_FIELD_CAP_TAG_RNDV_MAX_IOV          = UCS_BIT(28),
+
+    UCT_IFACE_ATTR_FIELD_CAP_ATOMIC32_OP_FLAGS         = UCS_BIT(29),
+    UCT_IFACE_ATTR_FIELD_CAP_ATOMIC32_FOP_FLAGS        = UCS_BIT(30),
+
+    UCT_IFACE_ATTR_FIELD_CAP_ATOMIC64_OP_FLAGS         = UCS_BIT(31),
+    UCT_IFACE_ATTR_FIELD_CAP_ATOMIC64_FOP_FLAGS        = UCS_BIT(32),
+
+    UCT_IFACE_ATTR_FIELD_CAP_FLAGS                     = UCS_BIT(33),
+    UCT_IFACE_ATTR_FIELD_CAP_EVENT_FLAGS               = UCS_BIT(34),
+
+    UCT_IFACE_ATTR_FIELD_DEVICE_ADDR_LEN               = UCS_BIT(35),
+    UCT_IFACE_ATTR_FIELD_IFACE_ADDR_LEN                = UCS_BIT(36),
+    UCT_IFACE_ATTR_FIELD_EP_ADDR_LEN                   = UCS_BIT(37),
+    UCT_IFACE_ATTR_FIELD_MAX_CONN_PRIV                 = UCS_BIT(38),
+    UCT_IFACE_ATTR_FIELD_LISTEN_SOCKADDR               = UCS_BIT(39),
+    UCT_IFACE_ATTR_FIELD_OVERHEAD                      = UCS_BIT(40),
+    UCT_IFACE_ATTR_FIELD_BANDWIDTH                     = UCS_BIT(41),
+    UCT_IFACE_ATTR_FIELD_LATENCY                       = UCS_BIT(42),
+    UCT_IFACE_ATTR_FIELD_PRIORITY                      = UCS_BIT(43),
+    UCT_IFACE_ATTR_FIELD_MAX_NUM_EPS                   = UCS_BIT(44),
+    UCT_IFACE_ATTR_FIELD_DEV_NUM_PATHS                 = UCS_BIT(45),
+
+} uct_iface_attr_field_t;
+
+#define UCT_IFACE_ATTR_V1_BITMASK UCS_MASK(46)
+
+typedef struct uct_iface_attr_v2 {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_iface_attr_field_t.
+     */
+    uint64_t          field_mask;
+
+    struct {
+        struct {
+            size_t           max_short;  /**< Maximal size for put_short */
+            size_t           max_bcopy;  /**< Maximal size for put_bcopy */
+            size_t           min_zcopy;  /**< Minimal size for put_zcopy (total
+                                              of @ref uct_iov_t::length of the
+                                              @a iov parameter) */
+            size_t           max_zcopy;  /**< Maximal size for put_zcopy (total
+                                              of @ref uct_iov_t::length of the
+                                              @a iov parameter) */
+            size_t           opt_zcopy_align; /**< Optimal alignment for zero-copy
+                                              buffer address */
+            size_t           align_mtu;       /**< MTU used for alignment */
+            size_t           max_iov;    /**< Maximal @a iovcnt parameter in
+                                              @ref ::uct_ep_put_zcopy
+                                              @anchor uct_iface_attr_cap_put_max_iov */
+        } put;                           /**< Attributes for PUT operations */
```

Comment:
uct_iface_op_attr_t *put;
uct_iface_op_attr_t contains own field_mask
and we have UCT_IFACE_ATTR_FIELD_PUT flag
same for get, am, tag.eager/rndv/recv